doc/sonic-build-system/build-enhancements.md

```diff
@@ -0,0 +1,463 @@
+	    -  Eg: docker run -v /var/run/docker.sock:/var/run/docker.sock ...
+	- When a new docker container/builder/composer is invoked from a build container:
+        - It is started as a sibiling to build container. 
+        - It will run in parallel with build container.
+	- This mode provides a better performance as it can utilize the full potential of host machine.
+
+### Build Container in SONiC:
+- The current SONiC build infrastructure generats all the SONiC build artifacts inside the docker container environment. When docker is isolated from the host CPU, the docker resource usage and filesystem access are restricted from its full capacity. Docker isolation is more essential for application containers, but for the build containers, the more essentail requirement is the build performace rather than adopting a higher security model. It provides the better build performance when the build containers are run in native mode. 
+- Sonic supports both the mode of build container creation. 
+- The Native docker mode gives better performace but it has some limitations:
+    - In a shared build servers, sonic docker creation from multiple user would give conflict as it shares the same docker image name.
+- This feature addresses:
+    - Sonic docker container creation in parallel from multiple users.
+	- Since it runs as sibiling container, it will degrade the parent container performace. 
+	- As it shares the host dockerd, it gives better performance as the multilevel UNIONFS/OverlayFS is not needed.
+
+#### Build Container in SONiC:
+
+
+![ Native Docker Support ](images/sonic-native-docker-support.png)
+
+
+- Currently, the build dockers are created as user dockers(docker-base-stretch-, etc) that are specific to each user. But the sonic dockers (docker-database, docker-swss, etc) are created with a fixed docker name and that are common to all the users.
+
+    - docker-database:latest
+    - docker-swss:latest
+
+- When multiple builds are triggered on the same build server that creates parallel building issue because all the build jobs are trying to create the same docker with latest tag. This happens only when sonic dockers are built using native host dockerd for sonic docker image creation.
+
+- This feature creates all sonic dockers as user sonic dockers and then, whilesaving and loading the user sonic dockers, it rename the user sonic dockers into correct sonic dockers with tag as latest.
+
+- The user sonic docker names are derived from '_LOAD_DOCKERS' make variable and using Jinja template, it replaces the FROM docker name with correct user sonic docker name for
+  loading and saving the docker image.
+
+- The template processing covers only for common dockers, Broadcom and VS platform dockers. For other vendor specific dockers, respective vendors need to add the support.
+
+
+## Version cache support
+
+### Version components
+
+- Sonic build downloads lots of component from external web which includes
+    - Source code files
```

Comment:
- These source codes are maintained by external users. 
- It can undergo a frequent update or release.
- Maintaining these source codes as part of SONiC repo, might add extra overhead.